dns/message.py

```diff
@@ -696,7 +711,96 @@ def _parse_special_rr_header(self, section, count, position,
 
 
 class QueryMessage(Message):
-    pass
+    def resolve_chaining(self):
+        """Follow the CNAME chain in the response to determine the answer
+        RRset.
+
+        Raises NotQueryResponse if the message is not a response.
+
+        Raises dns.message.ChainTooLong if the CNAME chain is too long.
+
+        Raises AnswerForNXDOMAIN if the rcode is NXDOMAIN but an answer was
+        found.
+
+        Raises dns.exception.FormError if the question count is not 1.
+
+        Returns a tuple (dns.name.Name, int, rrset) where the name is the
+        canonical name, the int is the minimized TTL, and rrset is their
+        answer RRset, which may be ``None`` if the chain was dangling or
+        the response is an NXDOMAIN.
+        """
+        if self.flags & dns.flags.QR == 0:
+            raise NotQueryResponse
+        if len(self.question) != 1:
+            raise dns.exception.FormError
+        question = self.question[0]
+        qname = question.name
+        min_ttl = -1
```

Comment:
This should probably set `min_ttl` to some large value, so that later code can do things like `min_ttl = std::min(min_ttl, rrset.ttl)` without extra conditionals.